packages/Interface/src/OperatorVector/DTK_FieldMultiVector.cpp

```diff
@@ -93,17 +94,18 @@ void FieldMultiVector::pullDataFromApplication()
 void FieldMultiVector::pushDataToApplication()
 {
     Teuchos::ArrayView<const SupportId> field_supports =
-	d_field->getLocalSupportIds();
-    Teuchos::ArrayRCP<Teuchos::ArrayRCP<const double> > vector_view =
-	this->get2dView();
+        d_field->getLocalSupportIds();
+
     int num_supports = field_supports.size();
     int dim = d_field->dimension();
-    for ( int n = 0; n < num_supports; ++n )
+
+    for ( int d = 0; d < dim; ++d )
     {
-	for ( int d = 0; d < dim; ++d )
-	{
-	    d_field->writeFieldData( field_supports[n], d, vector_view[d][n] );
-	}
+        Teuchos::ArrayRCP<double> vector_view = this->getDataNonConst(d);
+        for ( int n = 0; n < num_supports; ++n )
+        {
+            d_field->writeFieldData( field_supports[n], d, vector_view[n] );
+        }
```

Comment:
These changes related to accessing data in a `Tpetra::MultiVector` seem reasonable to me.